src/internal/connector/graph/service.go

```diff
@@ -120,3 +221,72 @@ type ContainerResolver interface {
 	// Items returns the containers in the cache.
 	Items() []CachedContainer
 }
+
+// ---------------------------------------------------------------------------
+// Client Middleware
+// ---------------------------------------------------------------------------
+
+// LoggingMiddleware can be used to log the http request sent by the graph client
+type LoggingMiddleware struct{}
+
+func (handler *LoggingMiddleware) Intercept(
+	pipeline khttp.Pipeline,
+	middlewareIndex int,
+	req *http.Request,
+) (*http.Response, error) {
+	var (
+		ctx       = req.Context()
+		resp, err = pipeline.Next(req, middlewareIndex)
+	)
+
+	if resp == nil {
+		return resp, err
+	}
+
+	if (resp.StatusCode / 100) == 2 {
```

Comment:
Add comment to identify to describe which errors are being caught in the 200 range

Reply:
I don't follow your statement.  The 200 range should not include errors.

Reply:
Apologies, not an error. Asking for comments on the different clauses there are few
`# Successful Response`

Reply:
Ah, gotcha. That's a good idea, thanks, I'll add it now.